doc/y_cable/design_doc.md

```diff
@@ -0,0 +1,240 @@
+## Multiple y cable vendor API design
+
+## Revision
+
+| Rev | Date     | Author          | Change Description |
+|:---:|:--------:|:---------------:|--------------------|
+| 0.1 | 03/22/21 | Vaibhav Dahiya  | Initial version    |
+
+## Scope
+
+### Overview
+
+This document summarizes the approach taken to accommodate multiple
+Y cable vendors for supporting dual-ToR configurations in SONiC.
+All the design choices are presented
+such that vendors can implement the Y cable interface with ease and effeciency
+and all the feature requirements are met.
+
+
+Challenge: to provide an interface for Y cable to interact with PMON docker and HOST
+           which can be uniform across all vendors.
+
+
+### Vendor to Module mapping
+
+#### Background
+
+- We need to have a vendor to Y cable module mappings defined and accessible somewhere for PMon container/cli to import
+
+  - Before calling any Y cable API, how do we know which type/vendor does the Y cable belong to ?
+  - Also lets suppose once the type/vendor of the cable is known from where does PMon container or cli import the module ?
+  - Since there can be many specs which vendors follow (8636, 8624 etc.) for transceivers it is important to have a solution which meets all requirements so that the API access is available to whoever wants to invoke it.
+  - another issue is how do we support the multiple vendors modules/packages which could be used by both SONiC PMon container docker as well as the sonic cli (sonic-utilities). Basically the package should be available both in host and container.
+#### Proposed Solution
+
+- We define a file which would contain vendor to module mapping and put it in a place which is accessible to both PMon container and host
+- The vendor to module mapping can be present in a file which could be present in sonic_platform_common
+- It makes sense to keep the mapping file in the sonic_y_cable package so that it can be updated in the same pull request when a vendor adds a new cable implementation. However, we cannot install data files outside of the Python directory via a wheel. Considering this we propose to make this a Python file which simply contains a 2D dictionary which can be used to look up the module path. If the file is part of the sonic_y_cable package, it will be installed in both the PMon container and the host
+
+For example
+```
+	/sonic_platform_common/sonic_y_cable/y_cable_vendor_mapping.py
+```
+
+Vendors can have several implementations/ways to use this concept
+- Vendor has a 1:1 mapping between model and Python file
+- Vendor has one Python file which supports all models
+- A combination of the above
+
+- Mapping could be such that its a dictionary in 2D
+```
+	<vendor_name>,<module>
+	<vendor_name>,<model_name>,<module>
+```
+- For example
+```
+	Credo,credo_model1,y_cable.py
+	Credo,y_cable.py
+```
```

Comment:
You updated the wording to reflect a 2D dictionary, but didn't update the examples to reflect.

Reply:
updated